python/raft-ann-bench/src/raft-ann-bench/run/conf/algos/faiss_cpu_ivf_flat.yaml

```diff
@@ -0,0 +1,21 @@
+name: faiss_cpu_ivf_flat
+groups:
+  base:
+    build:
+      nlist: [2048]
+      ratio: [1, 4, 10]
+      useFloat16: [False]
+      useRaft: [False]
+    search:
+      nprobe: [2048]
+      refine_ratio: [1]
+groups:
+  raft_enabed:
+    build:
+      nlist: [2048]
+      ratio: [1, 4, 10]
+      useFloat16: [False]
+      useRaft: [True]
+    search:
+      nprobe: [2048]
```

Comment:
Thanks for the comments. The StandardGpuResources object is indeed not thread safe according to the docs [here](https://github.com/facebookresearch/faiss/wiki/Threads-and-asynchronous-calls#:~:text=Faiss%20GPU%20indices,from%20different%20threads).
FAISS multi-threaded docs say that a separate GPU resource must be created for each thread. As such, sharing the same resource among separate threads, as we do in `raft-ann-bench` would not work. And creating a separate resource for each thread has the memory constraints as you mentioned. I can come to the conclusion that at this time we cannot support multi-threaded benchmarks for FAISS indices. I'll revert my comments referencing the FAISS issues and add a note explaining why we cannot support multi-threaded search.
I have confirmed that single-threaded benchmarks work in both -- latency and throughput mode.

Reply:
Thanks @tarang-jain and @tfeher for investigating this capability. Given this limitation is in FAISS and not RAFT/cuVS, and I suspect this might not be unique to just FAISS, I wonder if it would be worth adding an accessor on the indivudal raft-ann-bench index types to specify whether or not the index type can be used with the throughput mode (specifically with threads>1). This way, instead of a user encountering unexpected behavior, they can at least get an error that states the index type is not supported.